### PR TITLE
#18355 Fix: Unable to use future time machine snapshots

### PR DESCRIPTION
**Pull request by freddyucv**
TimeMachineFilter set the PageMode to PREVIEW but the Page Mode never get setting for Time Machine for this condition:

src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 1]
```diff
@@ -1472,4 +1476,53 @@ private static void addAnonymousPermissions(final Contentlet contentlet)
...
+                )
+                .request();
+
+
+        Mockito.when(mockRequest.getParameter("host_id")).thenReturn(site.getIdentifier());
+        mockRequest.setAttribute(WebKeys.HTMLPAGE_LANGUAGE, "1");
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(mockRequest);
+        final HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+
+        Calendar tomorrow = Calendar.getInstance();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1509]: Codacy Issue found: [Local variable 'tomorrow' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 2]
```diff
@@ -1472,4 +1476,53 @@ private static void addAnonymousPermissions(final Contentlet contentlet)
...
+        tomorrow.add(Calendar.DATE, 1);
+
+        //mockRequest.getSession().setAttribute("tm_date", Long.toString(tomorrow.getTime().getTime()));
+        //mockRequest.getSession().setAttribute("tm_lang", "1");
+
+        FilterChain chain = Mockito.mock(FilterChain.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1515]: Codacy Issue found: [Local variable 'chain' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 3]
```diff
@@ -15,6 +15,7 @@
 import com.dotcms.mock.request.MockAttributeRequest;
 import com.dotcms.mock.request.MockHttpRequest;
 import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.timemachine.ajax.TimeMachineAjaxAction;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.timemachine.ajax.TimeMachineAjaxAction'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1515]: Codacy Issue found: [Unnecessary use of fully qualified name 'Mockito.mock' due to existing static import 'org.mockito.Mockito.mock'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 4]
```diff
@@ -11,36 +13,41 @@
 
 import com.dotcms.api.web.HttpServletRequestThreadLocal;
 import com.dotcms.contenttype.model.type.ContentType;
-import com.dotcms.datagen.ContentletDataGen;
-import com.dotcms.datagen.FolderDataGen;
-import com.dotcms.datagen.HTMLPageDataGen;
-import com.dotcms.datagen.SiteDataGen;
-import com.dotcms.datagen.TemplateDataGen;
-import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:16]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 5]
```diff
@@ -1,6 +1,8 @@
 package com.dotcms.rendering.velocity.servlet;
 
 import static com.dotcms.datagen.TestDataUtils.getNewsLikeContentType;
+import static org.mockito.Matchers.any;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:4]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.Matchers.any'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 6]
```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
...
+        mockRequest.getSession().setAttribute("tm_lang", "1");
+
+        mockRequest.setAttribute(com.liferay.portal.util.WebKeys.USER, systemUser);
+        mockRequest.getSession().setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.EDIT_MODE);
+
+        FilterChain chain = Mockito.mock(FilterChain.class);
+        final TimeMachineFilter timeMachineFilter = new TimeMachineFilter();
+        timeMachineFilter.doFilter(mockRequest, mockResponse, chain);
+
+        VelocityServlet velocityServlet = new VelocityServlet();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:332]: Codacy Issue found: [Local variable 'velocityServlet' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 7]
```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
...
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(mockRequest);
+        mockRequest.getSession().setAttribute("tm_host" , host);
+        mockRequest.getSession().setAttribute(WebKeys.HTMLPAGE_LANGUAGE, "1");
+
+        final HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+
+        final ServletOutputStream outputStream = mock(ServletOutputStream.class);
+        when(mockResponse.getOutputStream()).thenReturn(outputStream);
+
+        Calendar tomorrow = Calendar.getInstance();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:318]: Codacy Issue found: [Local variable 'tomorrow' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 8]
```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
...
+        tomorrow.add(Calendar.DATE, 1);
+
+        //setting Time Machine
+        mockRequest.getSession().setAttribute("tm_date", Long.toString(tomorrow.getTime().getTime()));
+        mockRequest.getSession().setAttribute("tm_lang", "1");
+
+        mockRequest.setAttribute(com.liferay.portal.util.WebKeys.USER, systemUser);
+        mockRequest.getSession().setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.EDIT_MODE);
+
+        FilterChain chain = Mockito.mock(FilterChain.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:328]: Codacy Issue found: [Local variable 'chain' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 9]
```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
...
+                .setContentlet(contentlet)
+                .nextPersisted();
+
+        final HttpServletRequest mockRequest = new MockSessionRequest(
+                new MockAttributeRequest(new MockHttpRequest(host.getName(), ((HTMLPageAsset) page).getURI()).request()).request()
+        )
+                .request();
+
+
+        Mockito.when(mockRequest.getParameter("host_id")).thenReturn(host.getIdentifier());
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:307]: Codacy Issue found: [Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 10]
```diff
@@ -11,36 +13,41 @@
...
 import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
 import com.dotmarketing.portlets.languagesmanager.model.Language;
 import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Config;
 import com.dotmarketing.util.PageMode;
 import com.dotmarketing.util.WebKeys;
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
+import java.io.OutputStream;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:41]: Codacy Issue found: [Avoid unused imports such as 'java.io.OutputStream'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 11]
```diff
@@ -29,6 +31,7 @@
 import com.dotmarketing.exception.WebAssetException;
 import com.dotmarketing.factories.PublishFactory;
 import com.dotmarketing.factories.WebAssetFactory;
+import com.dotmarketing.filters.TimeMachineFilter;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:34]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.filters.TimeMachineFilter'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 12]
```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
...
+
+        Calendar tomorrow = Calendar.getInstance();
+        tomorrow.add(Calendar.DATE, 1);
+
+        //setting Time Machine
+        mockRequest.getSession().setAttribute("tm_date", Long.toString(tomorrow.getTime().getTime()));
+        mockRequest.getSession().setAttribute("tm_lang", "1");
+
+        mockRequest.setAttribute(com.liferay.portal.util.WebKeys.USER, systemUser);
+        mockRequest.getSession().setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.EDIT_MODE);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:326]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 13]
```diff
@@ -15,6 +15,8 @@
 import com.dotcms.mock.request.MockAttributeRequest;
 import com.dotcms.mock.request.MockHttpRequest;
 import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.rendering.velocity.servlet.VelocityServlet;
+import com.dotcms.timemachine.ajax.TimeMachineAjaxAction;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:19]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.timemachine.ajax.TimeMachineAjaxAction'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 14]
```diff
@@ -53,7 +56,9 @@
 import com.dotmarketing.util.WebKeys;
 import com.liferay.portal.model.User;
 import java.util.ArrayList;
+import java.util.Calendar;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:59]: Codacy Issue found: [Avoid unused imports such as 'java.util.Calendar'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 15]
```diff
@@ -15,6 +15,8 @@
 import com.dotcms.mock.request.MockAttributeRequest;
 import com.dotcms.mock.request.MockHttpRequest;
 import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.rendering.velocity.servlet.VelocityServlet;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.rendering.velocity.servlet.VelocityServlet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 16]
```diff
@@ -53,7 +56,9 @@
 import com.dotmarketing.util.WebKeys;
 import com.liferay.portal.model.User;
 import java.util.ArrayList;
+import java.util.Calendar;
 import java.util.List;
+import javax.servlet.FilterChain;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:61]: Codacy Issue found: [Avoid unused imports such as 'javax.servlet.FilterChain'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 17]
```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
...
+    public void testingTimeMachine () throws Exception{
+
+        final User systemUser = APILocator.systemUser();
+        final ContentType contentGenericType = APILocator.getContentTypeAPI(systemUser).find("webPageContent");
+
+        Container container = new ContainerDataGen().site(host).nextPersisted();
+        final Template template = new TemplateDataGen().site(host).withContainer(container.getIdentifier()).nextPersisted();
+
+        final List<ContainerStructure> csList = new ArrayList<ContainerStructure>();
+        final ContainerStructure cs = new ContainerStructure();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:272]: Codacy Issue found: [Avoid variables with short names like cs](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:328]: Codacy Issue found: [Unnecessary use of fully qualified name 'Mockito.mock' due to existing static import 'org.mockito.Mockito.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 18]
```diff
@@ -11,36 +12,40 @@
 
 import com.dotcms.api.web.HttpServletRequestThreadLocal;
 import com.dotcms.contenttype.model.type.ContentType;
-import com.dotcms.datagen.ContentletDataGen;
-import com.dotcms.datagen.FolderDataGen;
-import com.dotcms.datagen.HTMLPageDataGen;
-import com.dotcms.datagen.SiteDataGen;
-import com.dotcms.datagen.TemplateDataGen;
-import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:15]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)

**Review by erickgonzalez — APPROVED** [review]

**Comment by erickgonzalez** [comment]
Keep in mind the tests since some are failing but not sure if it is because of this change.

**Pull request merged**